### ISSUE TITLE
Markdown markup issues with Vim, but not Neovim

### ISSUE THREAD

**Issue by dixius99** [arcticicestudio/nord-vim#189]
I'm having trouble getting Markdown markup to display correctly in Vim, but it is working correctly in Neovim. My configs are identical.

Here are some screens to try to show what I mean. First, here's Vim:
<img width="493" alt="Screen Shot 2019-12-31 at 3 09 57 PM" src="https://user-images.githubusercontent.com/1924526/71633015-bd309300-2bdf-11ea-957f-1105629caa97.png">
And here's Neovim:
<img width="493" alt="Screen Shot 2019-12-31 at 3 10 14 PM" src="https://user-images.githubusercontent.com/1924526/71633020-c457a100-2bdf-11ea-80ab-ca9292694bb5.png">
You can see in the Vim screen that the second asterisk in the bold text is a different colour, and from that point on, all text is italic, regardless of what I do next. If I had used asterisks for the first italic word, instead of underscores, the closing asterisk for that word would be the different colour, and all text beyond that point would remain italic. 

Even though it is the same content, in the same terminal (iTerm 2), and with the same config, I can't seem to get the italics to display correctly when using Vim. Is there something I need to do so that Vim will behave the same as Neovim?

**Comment by arcticicestudio** [comment]
Hi @dixius99, I tried to reproduce the problem, but it works fine in both _Vim_ and _Neovim_ in terminal & GUI mode.

<p align="center">With <strong>enabled</strong> <code>nord_italic_font</code> option</p>
<p align="center"><img src="https://user-images.githubusercontent.com/7836623/71645525-39a79c80-2cda-11ea-8d59-e4e0f32d0199.png" width="80%" /></p>

<p align="center">With <strong>disabled</strong> <code>nord_italic_font</code> option</p>
<p align="center"><img src="https://user-images.githubusercontent.com/7836623/71645526-39a79c80-2cda-11ea-9d7f-5dd3e0100719.png" width="80%" /></p>

Could you please…

1. …try to reproduce the problem using a minimal _Vim_ configuration (clean `vimrc`), e.g. this one (using _vimplug_ as plugin manager):
  ```viml
call plug#begin(expand('~/.vim/plugged'))
Plug 'arcticicestudio/nord-vim'
call plug#end()

" Works with both enabled and disabled italic font option.
let g:nord_italic = 1

syntax enable
colorscheme nord
  ```
2. …make sure you're running the [latest plugin version][rll].
3. …try to reproduce the problem using a different theme (builtin or another theme plugin like e.g. _solarized_).
4. …try to reproduce the problem when enabling [Nord's italic font rendering option][dfi].
5. …make sure you're aware of the [requirements and tricky „hacks” when it comes to font rendering][dffr] in terminals and _Vim_.

[dfi]: https://www.nordtheme.com/docs/ports/vim/configuration#italic-style
[dffr]: https://www.nordtheme.com/docs/ports/vim/configuration#font-rendering
[rll]: https://github.com/arcticicestudio/nord-vim/releases/latest

**Comment by dixius99** [comment]
1. Same issue with minimal config
2. Yes, latest version per Vim Plug
3. **HOWEVER**, I seem to have the same issue with other themes that support italics.

So, there must be something up with my config. Not sure what that might be but it doesn't seem to be anything to do with Nord. I'll close the issue and use nvim until I can figure it out.

**Comment by dixius99** [comment]
After some messing around, I wanted to share that the issues appears to be related to my particular version of Vim. I have the latest version of Vim (8.2) installed on my Mac via `brew`, and that gives me the weird scenario I outlined above. If I use the bundled version fo Vim that comes with macOS Catalina (8.1.132), the problem goes away. I can't confirm if this is something to do with 8.2, or **my** 8.2, but I'm sharing just in case someone else stumbles on to the same problem and they end up searching for closed issues.

**Comment by arcticicestudio** [comment]
Thanks for the feedback, unfortunately implementation and rendering differences of systems and terminals are the root cause of such problems most of the time and it's almost impossible to document or provide workarounds for it.
Anyway, the problem might be related to the way _Vim_ gets compiled: The compilation [options used in the _Homebrew_ formula][fo] might differ from the one shipped with macOS.

You could also test if the problem is still occuring when you enable _true color_ support in Vim by setting the `termguicolors` option:

```viml
if (has("termguicolors"))
  set termguicolors
endif
```

This way _Vim_ in terminal mode will behave almost the same like in GUI mode regarding the font and colorspace rendering.

[fo]: https://github.com/Homebrew/homebrew-core/blob/master/Formula/vim.rb#L42-L55

**Comment by dixius99** [comment]
I'm not exactly sure if a close issue is the best way for me to communicate :), but I thought I would share with you that I discovered what was causing the inconsistency I shared above. I didn't have anything to do with the Homebrew formula options, but rather was related to differences in the `$VIMRUNTIME/syntax/markdown.vim` file for the various versions I tested with. Vim 8.2 uses a newer version of Tim Pope's `markdown.vim`, dated December 5, 2019, while most of the other versions of Vim I was using had a `markdown.vim` dated August 29, 2016.

Doing a diff of the two files, I can see several differences between them, including what I believe is how the various Markdown regions are defined.

If I just copy the old `markdown.vim` file into my new `$VIMRUNTIME`, the display issues are "fixed".

I wonder if this will become a bigger issue for theme writers in the future, as more people upgrade to the latest Vim.
